data-table/table.js

```diff
@@ -236,57 +258,115 @@ function cellIndexToElementId(wrapperDivId, rowIndex, colIndex) {
+
     /**
-     * Deletes row from existing table. Maybe should also
-     * accept index as an argument?
-     * TODO: Fill this out
+     * TODO: Implement (editable?) container for already existing row names
      */
-}
 
+    createRowInputAndBtn(config);
+
+    let br = document.createElement("br");
+    let br1 = document.createElement("br");
+    let br2 = document.createElement("br");
+
+    entryBoxDiv.appendChild(br);
+    entryBoxDiv.appendChild(br1);
+    entryBoxDiv.appendChild(br2);
+
+    createRowDeleteBtn(config);
+}
 
 /**
+ * TODO: Decide to keep/delete this function. Currently not used for anything.
  * This function takes the id of the list container for the candidates names
  * It then adds to the list another space for another candidate at the bottom of the list
- * @param wrapperDivId id of the list container
+ * @param {object} config   - Table configuration object
+ * @returns {undefined}     - Doesn't return anything
  */
-function addRow(wrapperDivId) {
-    let container = document.getElementById(wrapperDivId);
+function addRow(config) {
+    let container = document.getElementById(config.tableIds.entryBoxDivId);
     let child = document.createElement("DIV");
-    /** TODO: replace candidate's name below with config.Rowname */
-    child.innerHTML = 'Enter candidate\'s name'
+    child.innerHTML = "Enter " + config.datumConfig.names + ":";
     container.appendChild(child);
 }
 
 /**
- * This function enables the user to enter the name of the candidate,
- * When the container with the id is clicked, an input field is created that enables the user
- * to input the name of the candidate.
- * If the user focuses out of the input field, the name is updated appropriately, and the
- * input element removed
- * @param wrapperDivId the id of the container
+ * This function enables the user to enter the name of a row. Creates a field for text input
+ * as well as a button that sends input text to addSingleRow().
+ * @param {object} config   - Table configuration object
+ * @returns {undefined}     - Doesn't return anything
  */
-function enterRowValue(wrapperDivId) {
-    let div = document.getElementById(wrapperDivId);
+function createRowInputAndBtn(config) {
+    let entryBoxDiv = document.getElementById(config.tableIds.entryBoxDivId);
+
+    // Creates the field for text input
     let input = document.createElement("INPUT");
     input.type = 'text';
-    /** TODO: replace candidate's name below with config.Rowname */
-    input.placeholder = "Enter candidate's name";
-    div.innerHTML.innerHTML = input;
-    input.focusout = function () {
+    input.placeholder = "Enter " + config.datumConfig.names;
+
+    // If the user hits enter while in the text box, click the addRowBtn
+    input.addEventListener("keyup", function(event) {
+        event.preventDefault();
+        if (event.code === "Enter") {
+            addRowBtn.click();
+        }
+    })
```

Comment:
I'm worried about this explicit "Enter" key handling - seems like there are some edge cases where it could go wrong. (Perhaps on mobile it won't work?)

Then again, this is just a nicety - the button itself is below. So maybe it's fine like this.

I don't have any suggestions, just flagging my worry!

Reply:
Added a FIXME to draw attention to it. Will keep an eye out for different/better solutions.